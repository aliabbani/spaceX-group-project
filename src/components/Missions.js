/* eslint-disable camelcase */
import PropTypes from 'prop-types';
import '../stylesheet/missions.css';
import { useDispatch } from 'react-redux';
import { joinMission, leaveMission } from '../redux/reducers/dataReducers';

const Missions = (props) => {
  const dispatch = useDispatch();
  const {
    missions,
  } = props;

  return (
    <table className="table table-striped table-bordered">
      <thead>
        <tr>
          <th className="mission-row">Mission</th>
          <th className="description-row">Description</th>
          <th className="mission-row">Status</th>
          <th className="mission-row">{' '}</th>
        </tr>
      </thead>
      <tbody>
        {missions.map((mission) => (
          <tr key={mission.mission_id}>
            <th scope="row">{mission.mission_name}</th>
            <td>{mission.description}</td>
            <td className="align-middle text-center">
              {mission.reserved
                ? <span className="badge bg-primary ">ACTIVE MEMBER</span>
                : <span className="badge bg-secondary">NOT A MEMBER</span>}
            </td>
            <td className="align-middle">
              {mission.reserved === true ? (
                <button
                  type="button"
                  className="btn btn-outline-danger"
                  onClick={() => dispatch(leaveMission(mission.mission_id))}
                >
                  Leave Mission
                </button>
              ) : (
                <button
                  type="button"
                  className="btn btn-outline-secondary"
                  onClick={() => dispatch(joinMission(mission.mission_id))}
                >
                  Join Mission
                </button>
              ) }
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  );
};

Missions.propTypes = {
  missions: PropTypes.instanceOf(Array).isRequired,
};

export default Missions;
