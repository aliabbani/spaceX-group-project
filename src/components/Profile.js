import React from 'react';
import ProfileRocketsContainer from './ProfileRocketsContainer';
import ProfileMissions from './ProfileMissions';

const Profile = () => (
  <div>
    <ProfileRocketsContainer />
    <ProfileMissions />
  </div>
);

export default Profile;
