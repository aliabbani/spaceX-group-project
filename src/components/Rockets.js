import React from 'react';
import { v4 as uuidv4 } from 'uuid';
import propTypes from 'prop-types';
import RocketInfo from './RocketInfo';

const Rockets = (props) => {
  const { rockets } = props;
  const { rocketsTemp } = rockets;
  return (
    <div>
      {rocketsTemp.map((rocket) => (
        <RocketInfo rocket={rocket} key={uuidv4()} />
      ))}
    </div>
  );
};

Rockets.defaultProps = {
  rockets: {},
  rocketsTemp: {},
};
Rockets.propTypes = {
  rockets: propTypes.instanceOf(Object),
  rocketsTemp: propTypes.instanceOf(Object),
};

export default Rockets;
