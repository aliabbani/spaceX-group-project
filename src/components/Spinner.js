import loading from '../images/spinner.gif';

const Spinner = () => <div className="spinner-container"><img src={loading} alt="loading" className="spinner" /></div>;

export default Spinner;
