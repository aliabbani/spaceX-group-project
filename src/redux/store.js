import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import spacereducer from './reducers';

const reducer = combineReducers({
  spacereducer,
});

const store = createStore(
  reducer,
  applyMiddleware(thunk, logger),
);

export default store;
