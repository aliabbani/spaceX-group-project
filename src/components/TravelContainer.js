import { Route, Switch } from 'react-router-dom';
import Navbar from './Navbar';
import Rockets from './Rockets';
import Missions from './Missions';
import Profile from './Profile';
import NotMatch from '../pages/NotMatch';

function TravelContainer() {
  return (
    <>
      <Navbar />
      <Switch>
        <Route exact path="/">
          <Rockets />
        </Route>
        <Route path="/missions">
          <Missions />
        </Route>
        <Route path="/profile">
          <Profile />
        </Route>
        <Route path="*">
          <NotMatch />
        </Route>
      </Switch>
    </>
  );
}

export default TravelContainer;
