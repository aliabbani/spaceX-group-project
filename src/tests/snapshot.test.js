import React from 'react';
import renderer from 'react-test-renderer';
import { NavLink, BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Missions from '../components/Missions';
import Rockets from '../components/Rockets';
import Profile from '../components/Profile';
import Spinner from '../components/Spinner';
import TravelContainer from '../components/TravelContainer';
import store from '../redux/store';

describe('Component test snapshot', () => {
  it('App renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <TravelContainer />
          </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Router renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <NavLink to="/">Rockets</NavLink>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Spinner renders correctly', () => {
    const tree = renderer
      .create(
        <Spinner />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Rockets renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Missions renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Profile renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
