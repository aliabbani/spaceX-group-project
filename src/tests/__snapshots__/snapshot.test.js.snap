// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component test snapshot App renders correctly 1`] = `
Array [
  <nav
    className="navBar"
  >
    <div
      className="left-nav"
    >
      <img
        alt="logo"
        className="logo-nav"
        src="logo.png"
      />
      <h1
        className="main-title"
      >
        Space Travelers Hub
      </h1>
    </div>
    <ul
      className="right-elements"
    >
      <li
        className="right-nav-li"
      >
        <a
          aria-current="page"
          className="left-nav-a active-link"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Rockets
        </a>
      </li>
      <li
        className="right-nav-li"
      >
        <a
          aria-current={null}
          className="left-nav-a"
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="right-nav-li"
      >
        <a
          aria-current={null}
          className="left-nav-a"
          href="/profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="spinner-container"
  >
    <img
      alt="loading"
      className="spinner"
      src="spinner.gif"
    />
  </div>,
]
`;

exports[`Component test snapshot Missions renders correctly 1`] = `
<div
  className="main-missions px-5"
>
  <table
    className="table table-striped table-bordered"
  >
    <thead>
      <tr>
        <th
          className="mission-row w-15"
        >
          Mission
        </th>
        <th
          className="description-row w-55"
        >
          Description
        </th>
        <th
          className="mission-row w-15"
        >
          Status
        </th>
        <th
          className="mission-row w-15"
        >
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Component test snapshot Profile renders correctly 1`] = `
<div
  className="profile-container"
>
  <div
    style={
      Object {
        "maxWidth": "50%",
      }
    }
  >
    <h3>
       
      No rockets reserved!
    </h3>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="list-group"
          onKeyDown={[Function]}
          role="tablist"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="tab-content"
        />
      </div>
    </div>
  </div>
  <div
    className="mission-container"
  >
    <h3>
      No Missions joined!
    </h3>
  </div>
</div>
`;

exports[`Component test snapshot Rockets renders correctly 1`] = `
<div
  className="main-rocket"
/>
`;

exports[`Component test snapshot Router renders correctly 1`] = `
<a
  aria-current="page"
  className="active"
  href="/"
  onClick={[Function]}
  style={Object {}}
>
  Rockets
</a>
`;

exports[`Component test snapshot Spinner renders correctly 1`] = `
<div
  className="spinner-container"
>
  <img
    alt="loading"
    className="spinner"
    src="spinner.gif"
  />
</div>
`;
